lasagne/utils.py

```diff
@@ -21,6 +21,29 @@ def floatX(arr):
     return np.asarray(arr, dtype=theano.config.floatX)
 
 
+def no_upcast(list_element):
```

Comment:
Why not use `*args`?  This would save you from having to construct a list to call the function, which would make it cleaner, IMO:
```Python
def no_upcast(*args):
    result_list = []
     for element in args:
         if isinstance(element, np.ndarray):
             result_list.append(T.constant(element))
         else:
             result_list.append(element)
     return result_list
```
and the call signature would be e.g.
```
learning_rate, rho, epsilon = utils.no_upcast(
    learning_rate, rho, epsilon)
```

... or, better yet, make it a pretty one-liner with a list comprehension
```
def no_upcast(*args):
     return (T.constant(e) if isinstance(e, np.ndarray) else e for e in args)
```